modules/adlooxRtdProvider.js

```diff
@@ -213,193 +99,119 @@ function init(config, userConsent) {
     }
-  }
-
-  const refererInfo = getRefererInfo();
-  const args = [
-    [ 'v', `pbjs-${getGlobal().version}` ],
-    [ 'c', config.params.clientid ],
-    [ 'p', config.params.platformid ],
-    [ 't', config.params.tagid ],
-    [ 'imp', config.params.imps ],
-    [ 'fc_ip', config.params.freqcap_ip ],
-    [ 'fc_ipua', config.params.freqcap_ipua ],
-    [ 'pn', (refererInfo.page || '').substr(0, 300).split(/[?#]/)[0] ]
-  ];
-
-  if (!adUnits.length) {
-    logWarn(MODULE, 'no suitable adUnits (missing pbadslot?)');
-  }
-  const atfQueue = [];
-  adUnits.map((adUnit, i) => {
-    const ref = [ adUnit.gpid ];
-    if (!config.params.slotinpath) ref.push(adUnit.unit.code);
-    args.push(['s', ref.join('\t')]);
-
-    semaphoreInc(1);
-    atfQueue.push(function() {
-      atf(adUnit.unit, function(x) {
-        let viewable = document.visibilityState === undefined || document.visibilityState == 'visible';
-        try { viewable = viewable && top.document.hasFocus() } catch (_) {}
-        logInfo(MODULE, `atf code=${adUnit.unit.code} has area=${x}, viewable=${viewable}`);
-        const atfList = []; atfList[i] = { atf: parseInt(x * 100) };
-        response = mergeDeep(response, { _: atfList });
-        semaphoreInc(-1);
-      });
-    });
-  });
-  function atfCb() {
-    atfQueue.forEach(x => x());
-  }
-  if (document.readyState == 'loading') {
-    document.addEventListener('DOMContentLoaded', atfCb, false);
-  } else {
-    atfCb();
-  }
-
-  analyticsCommand(COMMAND.URL, {
-    url: (config.params.api_origin || API_ORIGIN) + '/q?',
-    args: args
-  }, function(url) {
-    ajax(url, {
-      success: function(responseText, q) {
-        try {
-          if (q.getResponseHeader('content-type') == 'application/json') {
-            response = mergeDeep(response, JSON.parse(responseText));
-          } else {
-            throw false;
-          }
-        } catch (_) {
-          logError(MODULE, 'unexpected response');
-        }
-        semaphoreInc(-1);
-      },
-      error: function(statusText, q) {
-        logError(MODULE, 'request failed');
-        semaphoreInc(-1);
-      }
-    });
-  });
+  );
 }
 
-function getTargetingData(adUnitArray, config, userConsent, auction) {
-  function targetingNormalise(v) {
+function getTargetingData(adUnitArray, config, userConsent) {
+  function val(v) {
     if (isArray(v) && v.length == 0) return undefined;
     if (isBoolean(v)) v = ~~v;
     if (!v) return undefined; // empty string and zero
     return v;
   }
 
-  const ortb2 = auction.getFPD().global || {};
-  const dataSite = deepAccess(ortb2, `site.ext.data.${MODULE_NAME}_rtd`) || {};
-  if (!dataSite.ok) return {};
+  const targeting = {};
 
-  const dataUser = deepAccess(ortb2, `user.ext.data.${MODULE_NAME}_rtd`) || {};
-  return getGlobal().adUnits.filter(adUnit => includes(adUnitArray, adUnit.code)).reduce((a, adUnit) => {
-    a[adUnit.code] = {};
+  const ortb2base = {};
+  _each(config._fpd, function(v0, k) {
```

Comment:
I broke this way back when - sorry! 

The issue is the `auction` argument is the "public" auction object exposed through events and not the internal one that has `getFPD`. 

Unfortunately I don't think this fix is correct, because if multiple auctions are running concurrently, they would conflict on the shared `config._fpd` handle.

You can get access to `getFPD` with:

`const ortb2 = auctionManager.index.getAuction(auction).getFPD().global || {}`

Or alternatively you could modify `config._fpd`  to be a map from auctionId to to FPD.

Reply:
> I broke this way back when - sorry!

Happens to the best of us, no sweat.

> You can get access to getFPD with:
> 
> const ortb2 = auctionManager.index.getAuction(auction).getFPD().global || {}

I prefer to do this, but I opt'ed to create `config._fpd` as 'auction' no longer (or maybe never was?) officially [passed in according to the documentation](https://docs.prebid.org/dev-docs/add-rtd-submodule.html#the-sub-module-object)

As no other RTD module seemed to use it I assumed you may be trying to get rid of it?

Reply:
I think that's the documentation getting out of sync - the argument was added in https://github.com/prebid/Prebid.js/pull/8119 and there are no plans to remove it as of now. 

In fact, digging a little more, I think the argument is necessary. Let me move down a few lines...